### ISSUE TITLE
Unpacking packed data from string

### ISSUE THREAD

**Issue by emilsoman** [creationix/msgpack-js-browser#13]
Seems like `msgpack.decode` assumes the encoded data is an instance of ArrayBuffer. Is there a way to unpack a JS string object containing packed data ? For example: 

```js
var packed = "\x81\xA3foo\xA3bar";
msgpack.unpack(packed); // => {foo: 'bar'}
```

**Comment by creationix** [comment]
Yes, the bodec library that's used can easily do this.  Assuming your string is encoded as "raw" encoding, the `fromRaw` utility is what you want. https://github.com/creationix/bodec/blob/master/bodec-browser.js#L200-L210

Note that strings in JavaScript are unicode and not 8-bit arrays.  In particular, they are 16-bit arrays (UCS-16).  The "raw" encoding is when you treat each 16-bits as a single byte and ignore the higher 8 bits.  JavaScript strings normally encode as UTF-8 when converting to raw 8-bit byte arrays so be careful when working with "raw" strings that use the right encoder when encoding to-from bytes.
